Header.c

```diff
@@ -95,26 +97,28 @@ MeterModeId Header_addMeterByName(Header* this, const char* name, int column) {
       char* end, dynamic[32] = {0};
       int ok = sscanf(paren, "(%10u)", &param); // CPUMeter
       if (!ok) {
-         ok = sscanf(paren, "(%30s)", dynamic); // DynamicMeter
-         if (ok && (end = strrchr(dynamic, ')'))) *end = '\0';
-         param = ok ? DynamicMeter_search(this->pl, dynamic) : 0;
+         if (sscanf(paren, "(%30s)", dynamic)) { // DynamicMeter
+            if ((end = strrchr(dynamic, ')')) == NULL)
+               return false;    // indicate htoprc parse failure
+            *end = '\0';
+            if (!DynamicMeter_search(this->pl->dynamicMeters, dynamic, &param))
+               return false;    // indicates name lookup failure
+         }
       }
       *paren = '\0';
    }
-   MeterModeId mode = TEXT_METERMODE;
    for (const MeterClass* const* type = Platform_meterTypes; *type; type++) {
       if (String_eq(name, (*type)->name)) {
          Meter* meter = Meter_new(this->pl, param, *type);
          Vector_add(meters, meter);
-         mode = meter->mode;
          break;
       }
    }
 
    if (paren)
       *paren = '(';
```

Comment:
Unnecessary assignment, subject to DSE?

Reply:
@BenBE excuse me, DSE refers to what?

Reply:
DSE = Dead Store Elimination; one of the compiler optimization steps.

Reply:
I think its needed and doing useful work - is it not? - it's supposed to reset the character in 'name' that we temporarily replaced with a NULL, so we could treat a sub-string as a regular null-terminated string.

If we don't do this we muck up the contents of colSettings->names[i] in Header_populateFromSettings, don't we?

FWIW, this code has been the forever as part of the CPUMeter handling of parenthesis - i.e. CPU(N) - in this routine.

Reply:
Having a closer look at this, there's a constness issue: We are handing in `const char*` but subsequently modifying that string. This should get some closer look+refactoring in another patch.

Reply:
Yep - the issue predates these changes by several years.